### PR TITLE
NMS-8142: Fix Collapsible Criteria not persisted

### PR DESCRIPTION
**Pull request by pdzaccone**
* Jira: https://issues.opennms.org/browse/NMS-8142

### CODE REVIEW

**Comment by opennms-bamboo** [comment]
Welcome @pdzaccone.

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username @pdzaccone.
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by opennms-bamboo** [comment]
Welcome @pdzaccone.

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username @pdzaccone.
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by opennms-bamboo** [comment]
Welcome @pdzaccone.

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username @pdzaccone.
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

File: features/topology-map/plugins/org.opennms.features.topology.plugins.topo.history/pom.xml [hunk 1]
```diff
@@ -57,6 +57,15 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest-library</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-all</artifactId>
+        </dependency>
```
Comment (mvrueden) [inline features/topology-map/plugins/org.opennms.features.topology.plugins.topo.history/pom.xml]: The scope of this dependency should be `test`.

File: features/topology-map/org.opennms.features.topology.app/src/main/java/org/opennms/features/topology/app/internal/AlarmSearchProvider.java [hunk 2]
```diff
@@ -200,9 +191,9 @@ public void setSeverityQuery(boolean isSeverityQuery) {
             String nodeLabel = alarm.getNodeLabel();
             LOG.debug("SearchProvider.query: adding '{}' to set of results.", nodeLabel);
 
-            AlarmSearchResult result = new AlarmSearchResult(alarm.getId(), alarm.getNodeId(), alarm.getNodeLabel(), queryString);
+            //TODO RS Find out when this method is used and check whether it is only called for collapsed nodes
```
Comment (mvrueden) [inline features/topology-map/org.opennms.features.topology.app/src/main/java/org/opennms/features/topology/app/internal/AlarmSearchProvider.java]: Please ensure that there are no TODOs in the code base created by you.

**Comment by mvrueden** [comment]
Thank you for your contribution 👍 

This looks very good from my point of view.
One thing to consider:
I am not a big fan of `boolean` constructor arguments. 
Especially if there are more. 
It makes the usage hard. 
Maybe there is a better way of setting the `collapsed` and `collapsible` properties.

I am going to give this a spin.

**Comment by opennms-bamboo** [comment]
Welcome @pdzaccone.

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username @pdzaccone.
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by mvrueden** [comment]
I just tested this and it works well.
I also pulled it to opennms to see what bamboo thinks about it: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS1461

**Comment by mvrueden** [comment]
Hey Ulf, can you please recheck the OCA?

**Comment by mvrueden** [comment]
Hey @pdzaccone can you please re-target this for `foundation-2017` instead?

**Comment by mvrueden** [comment]
This work needs to be rebased to `foundation-2017` before it can be merged.

**Pull request merged**